controllers/datamover.go

```diff
@@ -164,6 +166,61 @@ func (r *DPAReconciler) ReconcileDataMoverController(log logr.Logger) (bool, err
 	return true, nil
 }
 
+func (r *DPAReconciler) ReconcileDataMoverConfigs(log logr.Logger) (bool, error) {
+
+	// fetch latest DPA instance
+	dpa := oadpv1alpha1.DataProtectionApplication{}
+	if err := r.Get(r.Context, r.NamespacedName, &dpa); err != nil {
+		return false, err
+	}
+
+	// check configMap already exists
+	confMap, confMapExists, err := r.checkConfigMapExists()
+	if err != nil {
+		return false, err
+	}
+
+	// check for existing configMap but no data mover configs set
+	if !r.checkDataMoverConfigs(&dpa) && confMapExists {
```

Comment:
I believe we should delete the config map only if datamover is disabled by the user, right ?

Reply:
This deletes the configmap if the user deletes the config options on DPA, as then we would not want the VSB to use these options anymore. wdyt?